src/swap/SwapScreen.tsx

```diff
@@ -92,6 +106,33 @@ export function SwapScreen() {
     [swapAmount]
   )
 
+  const subTitleElement = useMemo(
+    () =>
+      exchangeRate && fromToken && toToken ? (
+        <Text style={[headerStyles.headerSubTitle, fetchingSwapQuote ? styles.mutedHeader : {}]}>
+          {`1 ${fromToken.symbol} ≈ ${new BigNumber(exchangeRate).toFormat(
+            5,
+            BigNumber.ROUND_DOWN
+          )} ${toToken.symbol}`}
+        </Text>
+      ) : undefined,
+    [exchangeRate, fromToken, toToken, fetchSwapQuoteError]
+  )
+
+  useLayoutEffect(() => {
```

Comment:
Would be nice to have this in the SwapScreenWithBack component, but tricky since this relies on local state variables and we'd have to move them to redux? If so, probably not worth it.

Reply:
Yeah I'd agree that it'd be preferable not to have this here, but I think moving some local state to redux is required.

Reply:
we actually have a task to remove the exchange rate from the header, this is the new design for the swap screen https://www.figma.com/file/jRtYI8GiSvPAaClBnRLLpu/In-App-Swap?node-id=2528-24667&t=HLZ14cPGl2sHRofV-0 - if we're creating extra complexity just to preserve the exchange rate in the header, i think it's not worth it :/ assuming that the exchange rate will not be in the header, this implementation becomes quite simple because we just need one prop to determine if the left icon is a back or a drawer icon right? it removes the need for these 2 screen wrappers? if that's the case i think i can just have a pass at the swap screen restyling tomorrow, it should be quick to knock out (famous last words 😅)

Reply:
I think we'd still need the screen wrappers because (1) drawer navigator components cannot accept props and (2) one of the cohorts in this experiment will not render the swap screen in the drawer, so the home actions cannot navigate to the swap screen unless we add it to the regular navigator. Joe explains this issue a bit more in #3610, and this PR mimics that approach

Reply:
yeah we still need the 2 screens, but sounds like we can drop the subtitle stuff and all the messy passing around of navigation props.

Reply:
sorry i don't really understand the problem fully (even after reading Joe's comment in #3610). screens registered in the normal navigator can accept props, it's only the drawer navigator screens that cannot? i didn't know this, but i don't see how this is a blocking problem here (could just make all props optional, and pass the necessary props through the normal navigator?) we currently have one screen `Screens.BackupIntroduction` which maybe has the same use case, it appears conditionally in the drawer menu for some users and for others it is a normal screen that we navigate to from the settings screen. i might be missing some details about why we need a different approach for these home nav cards, so ignore this suggestion if that's the case :) tagging @jophish in case he has some more to add

Reply:
either way, you can move forward in this PR with the assumption that we no longer need to display the exchange rate in the header. that should simplify at least the props and how the header is created :)

Reply:
Yeah I think that the reason the `BackupIntroduction` screen doesn't have this issue is because there is a `ts-expect-error` comment [added here](https://github.com/valora-inc/wallet/blob/main/src/navigator/DrawerNavigator.tsx#L303) in the `DrawerNavigator`. If we add that, it would remove the need for the two screens but not sure that we want to ignore that error.

Also, thanks for speeding through that exchange rate in header PR :)

Reply:
Thanks @kathaypacific , with regards to the 2 screens, the primary reason is the 2nd one mentioned by @anthonydaoud . We can workaround the ts error, but for one of the variants, we don't want to show the swap screen on the drawer menu, so it won't be registered with the drawer navigator and the action button on home screen won't be able to navigate to the screen. This is unlike the BackupIntroduction screen, which is also behind flags, but we don't want to navigate to it from any other place too if those flags are false.